### PR TITLE
Add word pieces tokenization script

### PR DESCRIPTION
**Pull request by yuki617**
* Add word pieces tokenization script for msmarco passages/document
* Add word pieces tokenization script for msmarco query

### CODE REVIEW

File: scripts/tokenization_document.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+add fields to jsonl with contents((BERT token)
+"""
+
+
+parser = argparse.ArgumentParser(description='Convert MSMARCO-adhoc documents.')
+parser.add_argument('--input', metavar='input file', help='input file',
+                    type=str, required=True)
+parser.add_argument('--output', metavar='output file', help='output file',
+                    type=str, required=True)
+parser.add_argument('--proc_qty', metavar='# of processes', help='# of NLP processes to span',
```
Comment (lintool) [inline scripts/tokenization_document.py:34]: how about an argument that's more intuitive? e.g., `--workers`?

File: scripts/tokenization_document.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#
```
Comment (lintool) [inline scripts/tokenization_document.py:1]: script names should begin with verbs - i.e., they're actions.

how about `tokenize_corpus`?

File: scripts/tokenization_query.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#
```
Comment (lintool) [inline scripts/tokenization_query.py:1]: `tokenize_queries`?

File: scripts/tokenization_query.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+add fields to query json with contents((BERT token)                                  
+"""
+
+
+parser = argparse.ArgumentParser(description='Convert MSMARCO-adhoc documents.')
+parser.add_argument('--input', metavar='input file', help='input file',
+                    type=str, required=True)
+parser.add_argument('--output', metavar='output file', help='output file',
+                    type=str, required=True)
+parser.add_argument('--proc_qty', metavar='# of processes', help='# of NLP processes to span',
```
Comment (lintool) [inline scripts/tokenization_query.py:34]: you don't need multiprocessing here, right?
simplify... fewer opportunities for bugs...

File: scripts/tokenization_query.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    type=int, default=multiprocessing.cpu_count() - 2)
+
+
+args = parser.parse_args()
+print(args)
+arg_vars = vars(args)
+
+inpFile = open(args.input)
+outFile = open(args.output, 'w')
```
Comment (lintool) [inline scripts/tokenization_query.py:43]: use `with` Python idiom? so you don't have worry about closing handles manually, etc.

File: scripts/tokenization_query.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    did, query = fields
+
+    doc = {"id": did,
+          "contents": get_retokenized(bert_tokenizer, query.lower())}
+    docStr = doc['id']+'\t'+doc['contents']+'\n'
+    outFile.write(docStr)
+
+    if line_num % 10000 == 0:
+        print('Processed %d queries' % line_num)
```
Comment (lintool) [inline scripts/tokenization_query.py:67]: use f-strings - throughout

**Review by lintool — CHANGES_REQUESTED** [review]

File: scripts/tokenization_document.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                    type=str, required=True)
+parser.add_argument('--proc_qty', metavar='# of processes', help='# of NLP processes to span',
+                    type=int, default=multiprocessing.cpu_count() - 2)
+
+
+args = parser.parse_args()
+print(args)
+arg_vars = vars(args)
+
+inpFile = open(args.input)
```
Comment (lintool) [inline scripts/tokenization_document.py:42]: use Python `with` idiom?

File: scripts/tokenize_corpus.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        if len(res) % 1000 == 0:
+            end = time.time()
+            print(f"finish {len(res)} using {end-start}")
+            start = end
+    return res
+
+
+if __name__ == '__main__':
+    workers = args.workers
+    print(f"Spanning {workers} processes")
```
Comment (lintool) [inline scripts/tokenize_corpus.py:81]: I think you mean "spawning"?

File: scripts/tokenize_queries.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import argparse
+from transformers import AutoTokenizer
+from tqdm import tqdm
+from ltr_msmarco.convert_common import get_retokenized
+"""
+add fields to query json with contents((BERT token)                                  
+"""
+
+
+parser = argparse.ArgumentParser(description='Convert MSMARCO-adhoc documents.')
```
Comment (lintool) [inline scripts/tokenize_queries.py:28]: queries in tsv format? (_not_ documents)

File: scripts/tokenize_queries.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+parser.add_argument('--output', metavar='output file', help='output file',
+                    type=str, required=True)
+
+args = parser.parse_args()
+print(args)
+arg_vars = vars(args)
+
+bert_tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+
+with open(args.input) as inpFile:
```
Comment (lintool) [inline scripts/tokenize_queries.py:40]: inpFile -> inFile?

File: scripts/tokenize_corpus.py [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+add fields to jsonl with contents((BERT token)
+"""
+
+
+parser = argparse.ArgumentParser(description='Convert MSMARCO-adhoc documents.')
+parser.add_argument('--input', metavar='input file', help='input file',
+                    type=str, required=True)
+parser.add_argument('--output', metavar='output file', help='output file',
+                    type=str, required=True)
+parser.add_argument('--workers', metavar='# of processes', help='# of NLP processes to span',
```
Comment (lintool) [inline scripts/tokenize_corpus.py:34]: why NLP here? just "# of workers to spawn"?

File: scripts/tokenize_corpus.py [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+import argparse
+import multiprocessing
+import json
+import time
+from joblib import Parallel, delayed
+from transformers import AutoTokenizer
+from ltr_msmarco.convert_common import get_retokenized
+
+"""
+add fields to jsonl with contents((BERT token)
```
Comment (lintool) [inline scripts/tokenize_corpus.py:25]: inaccurate...

**Review by lintool — APPROVED** [review]
minor tweaks, fix and then merge.

File: scripts/tokenize_corpus.py (hunk 12 above)
Comment (yuki617) [reply scripts/tokenize_corpus.py]: I have changed it into "Replace original contents fields with bert tokenization". Is it more accurate?

Comment (lintool) [reply scripts/tokenize_corpus.py]: yup

File: scripts/tokenize_corpus.py [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+        if not line:
+            return None
+        json_line = json.loads(line)
+        pid = json_line['id']
+        body = json_line['contents']
+
+        doc = {"id": pid,
+               "contents": get_retokenized(bert_tokenizer, body.lower())}
+        return doc
+    res = []
```
Comment (lintool) [inline scripts/tokenize_corpus.py:68]: for clarity, you might want an empty line here.

File: scripts/tokenize_corpus.py [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+if __name__ == '__main__':
+    workers = args.workers
+    print(f"Spawning {workers} processes")
+    pool = Parallel(n_jobs=workers, verbose=10)
+    line_num = 0
+
+    with open(args.input) as inFile:
+        with open(args.output, 'w') as outFile:
+
```
Comment (lintool) [inline scripts/tokenize_corpus.py:87]: for clarity, I think you want to kill this blank line?

File: scripts/tokenize_queries.py [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+# limitations under the License.
+#
+
+"""Convert MSMARCO queries"""
+
+import argparse
+from transformers import AutoTokenizer
+from tqdm import tqdm
+from ltr_msmarco.convert_common import get_retokenized
+"""
```
Comment (lintool) [inline scripts/tokenize_queries.py:23]: you don't need both this `"""` string here and above.

**Comment by lintool** [comment]
Final set of nits, please take a look, and then go ahead and merge.

**Comment by yuki617** [comment]
> Final set of nits, please take a look, and then go ahead and merge.

Resolved all comments mentioned above, will merge it soon!

**Pull request merged**